ipc/mdap.py

```diff
@@ -27,37 +28,67 @@ def connect_to_telemetry_server(address: tuple) -> socket.socket:
     print("Connection with the telemetry server established successfully.")
     return client_socket
 
+
 def telem_to_message(data_to_send, payloads):
     for telemetry in payloads[TelemetryPayload]:
         telemetry = cast(TelemetryPayload, telemetry)
-        data_to_send[f"{telemetry.module.name}-{telemetry.channel.name}"] = {
-            "value": telemetry.data,
-            "timestamp": int(time.time()),
-            "bucket": "mission_data"
-        }
 
-        if (telemetry.module.name, telemetry.channel.name) == ('WatchdogHeartbeatTvac', 'AdcTempKelvin'):
-            app.logger.notice(f"BATTERY TEMP IS: {telemetry.data - 273.15}ºC")
-            data_to_send["TempC"] = {
-                "value": telemetry.data - 273.15,
-                "timestamp": int(time.time()),
-                "bucket": "mission_data"
-            }
+        populate_data_to_send(telemetry.module.name, telemetry.channel.name,
+                              telemetry.data, data_to_send)
+
+        # TODO: The if condition below is solely for local testing purposes. Remove it later.
+        if telemetry.module.name == "WatchdogHeartbeatTvac":
+            # Convert the temperature from Kelvin to Celsius.
+            temperature = telemetry.data - 273.15
+            app.logger.notice(f"BATTERY TEMP IS: {temperature}ºC")
+
+            timestamp = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
+            if "WatchdogHeartbeatTvac" in data_to_send:
+                if "AdcTempKelvin" in data_to_send["WatchdogHeartbeatTvac"]["timestamp"]:
+                    data_to_send["WatchdogHeartbeatTvac"][timestamp]["AdcTempKelvin"] = temperature
+            else:
+                data_to_send["data"]["WatchdogHeartbeatTvac"] = {
+                    timestamp: {
+                        "AdcTempKelvin": temperature
+                    }
+                }
 
     return data_to_send
 
+
 def events_to_message(data_to_send, payloads):
     for event in payloads[EventPayload]:
         event = cast(EventPayload, event)
+        populate_data_to_send(event.module.name, event.event.name, event.formatted_string, data_to_send)
 
-        data_to_send[f"{event.module.name}-{event.event.name}"] = {
-            "value": event.formatted_string,
-            "timestamp": int(time.time()),
-            "bucket": "mission_events"
-        }
-    
     return data_to_send
 
+
+def populate_data_to_send(key, channel, data, data_to_send):
+    # Get the current time with microsecond precision
+    current_time = datetime.utcnow()
+    # Convert datetime to string with a specific format
+    timestamp = current_time.strftime("%Y-%m-%d %H:%M:%S")
+
+    if key in data_to_send["data"]:
+        # module exists
+        if timestamp in data_to_send["data"][key]:
+            # timestamp exists, update or add the channel and value pair
+            data_to_send["data"][key][timestamp][channel] = data
```

Comment:
You can use this kind of logic here to cleanup:

dictionary[<key>] = dictionary.get(<key>, <default value if not found>)